### PR TITLE
Create Samtools 1.7.0 Dockerfile:  Dockerfile uses "ARG software_version=1.7.0" to simplify updating to newer versions

### PR DESCRIPTION
**Pull request by alanhoyle**
# Submitting a Container
 
 (If you're requesting for a new container, please check the procedure described [here](https://github.com/BioContainers/containers#241-how-to-request-a-container).  
 
 ## Check BioContainers' Dockerfile [specifications](https://github.com/BioContainers/specs)  
 ## Checklist
 
 1. Misc
 - [ ] My tool doesn't exist in [BioConda](#make-sure-your-tool-isnt-already-present-in-bioconda)   
 - [ ] The image can be built   
 
 2. [Metadata](#check-biocontainers-dockerfile-metadata)  
 - [ ] LABEL base_image  
 - [ ] LABEL version
 - [ ] LABEL software.version  
 - [ ] LABEL about.summary  
 - [ ] LABEL about.home  
 - [ ] LABEL about.license   
 - [ ] MAINTAINER  
 
 3. Extra (optionals)
  - [ ] I have written tests in test-cmds.txt  
  - [ ] LABEL extra.identifier    
  - [ ] LABEL about.documentation   
  - [ ] LABEL about.license_file
  - [ ] LABEL about.tags   
 
 
 ## Check [BioContainers'](https://github.com/BioContainers/specs) Dockerfile metadata

### CODE REVIEW

**Comment by alanhoyle** [comment]
The build problem seems to be a problem with the jenkins spec check.  

In my Dockerfile, rather than having the lines:

```
LABEL software.version="1.3.1"
[... and ...]
RUN conda install samtools=1.3.1
```

I have the lines:

```
ARG software_version=1.7.0
[...]
LABEL software.version="$software_version"
[...]
RUN conda install ncurses samtools=$software_version
```

This successfully builds in a real docker environment.  It also adds two improvements (in my opinion):

1.  If a user has the Dockerfile and wants to build a new version, they only need to change the version to build once on the `ARG software_version=...` line instead of having to do it on both the `LABEL` line and the `RUN conda install...` line.  

1. If the user has the Dockerfile and wants to build a new version without editing the source, they can add `--build-arg software_version=...` to the `docker build ...` command.

**Comment by osallou** [comment]
The pb is the docker parse li rary does not interpret the arg.
O have to see how to workaround this

File: samtools/1.7.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+################## BASE IMAGE ######################
+
+FROM biocontainers/biocontainers:latest
+
+################## METADATA ######################
+
+ARG software_version=1.7.0
+LABEL base_image="biocontainers:latest"
+LABEL version="2"
+LABEL software="Samtools"
```
Comment (osallou) [inline samtools/1.7.0/Dockerfile]: software name must match regexp [a-z0-9_-] (lowercase) as name is used for container naming and docker names have restrictions

**Comment by osallou** [comment]
Found an existing bioconda package for this software (https://bioconda.github.io/recipes/samtools/README.html), is this the same, then you should update the recipe in bioconda to avoid duplicates.

**Comment by osallou** [comment]
do not take into accuont error about software.version. We tried a fix for ARG support, but seems not up-to-date yet

**Comment by osallou** [comment]
Found an existing bioconda package for this software (https://bioconda.github.io/recipes/samtools/README.html), is this the same, then you should update the recipe in bioconda to avoid duplicates.

**Comment by osallou** [comment]
No samtools/1.7.0/test-cmds.txt (test file) present, skipping tests

**Comment by osallou** [comment]
Found an existing bioconda package for this software (https://bioconda.github.io/recipes/samtools/README.html), is this the same, then you should update the recipe in bioconda to avoid duplicates.

**Comment by osallou** [comment]
No samtools/1.7.0/test-cmds.txt (test file) present, skipping tests

**Pull request merged**